### PR TITLE
feat: add sentence analyzer project

### PR DESCRIPTION
**Pull request by Ksound22**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

**Review by jdwilkin4 — COMMENTED** [review]
I went through the project and we are going to need to make some updates to the original prototype.

Originally, we weren't working with functions because the functions module was further down. But the labs team were having issues with testing and it made more sense to have function appear earlier. So we moved it.

As result, this project will work better from a testing perspective and learning perspective to use functions. 

Something along these lines:

```js
// this is just one way to solve this. Encourage campers to design the function their own way, with whatever loops they want, as long as they return the correct result

// or just have one function called getCount that takes in a strType and sentence since vowelCount, consonant and punctuationCount are all the same
function getVowelCount(sentence) {
  const vowels = "aeiou";
  let count = 0;

  for (const char of sentence.toLowerCase()) {
    if (vowels.includes(char)) {
      count++;
    }
  }
  return count;
}

// have a step to encourage campers to test out their function

const vowelCount = getVowelCount("Apples are tasty fruits");
console.log(`Vowel Count: ${vowelCount}`);

// this is just one way to solve this. Encourage campers to design the function their own way, with whatever loops they want, as long as they return the correct result
function getWordCount(sentence) {
  const words = sentence.trim().split(/\s+/);
  return words.length;
}

// have a step to encourage campers to test out their function

const wordCount = getWordCount("I love freeCodeCamp");
console.log(`Word Count: ${wordCount}`);

etc....
 
```

This will make it easier to test because we will only care about the final function outputs and test that. Also, it will better align with the camper feedback and moderator feedback we have gotten over the past several months. 

We should encourage them to design these functions using whatever loops and methods necessary. This will force campers to think more about what to do and allow for more creativity in solutions. This will also create a better learning experience. 

For example, one of the steps could be something like this:

```md
In this workshop, you will build a sentence analyzer that will take a sentence and get the count for the number of words, vowels, consonants, and punctuation marks. 

In this first step, create a `getVowelCount` function with parameter called `sentence`. Your function should return the total number of vowels in a sentence. 

For example, if the sentence is `"Apples are tasty fruits"` the function should return the number `7`.

A vowel is any of the following characters: `"aeiou"`. You will want to loop through the sentence and check if the current character is a vowel. You will need a way to keep track of the current count of vowels, and return that count.
```

**Review by Sembauke — APPROVED** [review]

File: curriculum/challenges/english/25-front-end-development/workshop-sentence-analyzer/66e2edb23013ca596f8d9568.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+assert.isNotNull(punctuationCount)
+```
+
+Your `punctuationCount` variable should be set to the result of `getPunctuationCount("WHAT?!?!?!?!?")`.
+
+```js
+assert.equal(punctuationCount, getPunctuationCount("WHAT?!?!?!?!?"));
+```
+
+You should log the `punctuationCount` variable to the console with the parameter `"Punctuation Count: ${punctuationCount}"`.
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/workshop-sentence-analyzer/66e2edb23013ca596f8d9568.md:28]: this hint of `"Punctuation Count: ${punctuationCount}"`  implies that we are only going to accept template literals here

```suggestion
You should log the following to the console: `"Punctuation Count: [Punctuation count goes here]"`. Replace  `[Punctuation count goes here]` with the the actual variable name. Make sure to use proper string concatenation syntax here.
```

File: curriculum/challenges/english/25-front-end-development/workshop-sentence-analyzer/66e2eab8a5638f57b637b7cc.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
+---
+id: 66e2eab8a5638f57b637b7cc
+title: Step 5
+challengeType: 1
+dashedName: step-5
+---
+
+# --description--
+
+You should count the number of puctuations now.
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/workshop-sentence-analyzer/66e2eab8a5638f57b637b7cc.md:10]:
```suggestion
You should count the number of punctuations now.
```

File: curriculum/challenges/english/25-front-end-development/workshop-sentence-analyzer/66e2f376df6f315ee81de81a.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+title: Step 8
+challengeType: 1
+dashedName: step-8
+---
+
+# --description--
+
+Test your `getWordCount` by creating a `wordCount` variable set to the calling of the `getWordCount` function with the sentence `"I love freeCodeCamp"`.
+
+After that, log the `wordCount` variable to the console with the text `"Word Count:"` and the interpolation of the `wordCount` variable. You can also use plus (`+`) concatenator if you want.
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/workshop-sentence-analyzer/66e2f376df6f315ee81de81a.md:12]:
```suggestion
After that, log the following to the console: `"Word Count: [Word count goes here]"`. Replace  `[Word count goes here]` with the the actual variable name. You can choose to use template strings or string concatenation with the `+` operator here.
```

**Review by jdwilkin4 — COMMENTED** [review]
Left some comments for rewords and fixing some spelling errors.
But otherwise looks good.

We might still get push back on the repetitive punctuation, vowel and consonant count functions since they are basically the same. But we will see how the community responds when this is in beta and make any adjustments from there

**Review by jdwilkin4 — APPROVED** [review]

**Pull request merged**